pkg/plugin/plugin_test.go

```diff
@@ -20,211 +20,52 @@ func TestCreateOrUpdate(t *testing.T) {
 	tests := []struct {
 		name     string
 		isUpdate bool
-		wantErr  bool
-		errStep  api.PluginStep
 	}{
 		{
 			name:     "deploy",
 			isUpdate: false,
-			wantErr:  false,
 		},
 		{
 			name:     "update",
 			isUpdate: true,
-			wantErr:  false,
 		},
-		{
-			name:     "deploy: deploy error",
-			isUpdate: false,
-			wantErr:  true,
-			errStep:  api.PluginStepDeploy,
-		},
-		{
-			name:     "deploy: initialize error",
-			isUpdate: false,
-			wantErr:  true,
-			errStep:  api.PluginStepInitialize,
```

Comment:
These tests were not very useful, so removing. the mocking below confirms the steps that we want to take for each flow.